CHANGELOG.md

```diff
@@ -1,5 +1,36 @@
 ## 5.42.0 (Unreleased)
 
+DEPRECATIONS:
+* compute: setting `google_compute_subnetwork.secondary_ip_range = []` to explicitly set a list of empty objects is deprecated and will produce an error in the upcoming major release. Use `send_secondary_ip_range_if_empty` while removing `secondary_ip_range` from config instead. ([#7961](https://github.com/hashicorp/terraform-provider-google-beta/pull/7961))
+
+
+FEATURES:
+* **New Data Source:** `google_artifact_registry_locations` ([#7922](https://github.com/hashicorp/terraform-provider-google-beta/pull/7922))
+* **New Data Source:** `google_cloud_identity_transitive_group_memberships` ([#7917](https://github.com/hashicorp/terraform-provider-google-beta/pull/7917))
+* **New Resource:** `google_discovery_engine_schema` ([#7963](https://github.com/hashicorp/terraform-provider-google-beta/pull/7963))
+* **New Resource:** `google_scc_folder_notification_config` ([#7928](https://github.com/hashicorp/terraform-provider-google-beta/pull/7928))
+* **New Resource:** `google_scc_v2_folder_notification_config` ([#7927](https://github.com/hashicorp/terraform-provider-google-beta/pull/7927))
+* **New Resource:** `google_vertex_ai_index_endpoint_deployed_index` ([#7931](https://github.com/hashicorp/terraform-provider-google-beta/pull/7931))
+
+IMPROVEMENTS:
+* clouddeploy: added `serial_pipeline.stages.strategy.canary.runtime_config.kubernetes.gateway_service_mesh.pod_selector_label` and `serial_pipeline.stages.strategy.canary.runtime_config.kubernetes.service_networking.pod_selector_label` fields to `google_clouddeploy_delivery_pipeline` resource ([#7945](https://github.com/hashicorp/terraform-provider-google-beta/pull/7945))
+* compute: added `TDX` instance option to `confidential_instance_type` instance in `google_compute_instance` ([#7913](https://github.com/hashicorp/terraform-provider-google-beta/pull/7913))
+* compute: added `send_secondary_ip_range_if_empty` to `google_compute_subnetwork` ([#7961](https://github.com/hashicorp/terraform-provider-google-beta/pull/7961))
+* dataflow: made provider return more descriptive errors when the `parameters` field of `google_dataflow_flex_template_job` contains Dataflow options ([#7943](https://github.com/hashicorp/terraform-provider-google-beta/pull/7943))
```

Comment:
```suggestion
```
This change is reverted from the release branch as it caused a couple of test failures in nightly.